### PR TITLE
Review/jm final

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
This PR serves as the final review of the Amplitude package. Please feel free to merge the changes from this PR into `main` and please address any comments within the notes below.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,16 +25,14 @@
 To use this dbt package, you must have the following:
 - At least one Fivetran Amplitude connector syncing data into your destination. 
 - A **BigQuery**, **Snowflake**, **Redshift**, **PostgreSQL**, or **Databricks** destination.
-  - If you are using Databricks you'll need to add the below to your `dbt_project.yml`. 
 
+### Databricks Dispatch Configuration
+If you are using a Databricks destination with this package you will need to add the below (or a variation of the below) dispatch configuration within your `dbt_project.yml`. This is required in order for the package to accurately search for macros within the `dbt-labs/spark_utils` then the `dbt-labs/dbt_utils` packages respectively.
```
Comment (fivetran-joemarkiewicz) [inline README.md:30]: Following the procedure that we have in place for the Ad Reporting package, just want to call this out in its own section

File: README.md [hunk 2]
```diff
@@ -25,16 +25,14 @@
...
 ```yml
-# dbt_project.yml
-
 dispatch:
   - macro_namespace: dbt_utils
     search_order: ['spark_utils', 'dbt_utils']
 ```
-- **dbt Version**: This dbt package requires you have a functional dbt project that utilizes a dbt version within the respective range `>=1.0.0, <2.0.0`.
```
Comment (fivetran-joemarkiewicz) [inline README.md:37]: Since we have the readme badge, this is not needed.

File: models/stg_amplitude.yml [hunk 3]
```diff
@@ -6,53 +6,52 @@ models:
 
     columns:
       - name: unique_event_id
-        description: A unique key for each event, concatenating event_id, device_id, and client_event_time
+        description: A unique key for each event, concatenating event_id, device_id, client_event_time, and amplitude_user_id.
```
Comment (fivetran-joemarkiewicz) [inline models/stg_amplitude.yml:9]: You also include `amplitude_user_id` in this unique key. Wanted to make sure we call that out.

File: models/stg_amplitude.yml [hunk 4]
```diff
@@ -6,53 +6,52 @@ models:
...
         tests:
           - unique
           - not_null
       - name: event_id
-        description: A counter that distinguishes events
+        description: A counter that distinguishes events.
```
Comment (fivetran-joemarkiewicz) [inline models/stg_amplitude.yml:14]: Let's be consistent with periods in the descriptions. Made this along with other documentation updates below.

File: models/stg_amplitude__event.sql [hunk 5]
```diff
@@ -71,9 +71,15 @@ final as (
...
+        {{ dbt_utils.surrogate_key(['event_id','device_id','client_event_time','amplitude_user_id']) }} as unique_event_id
+    from final
 )
 
-select
-    *,
-    {{ dbt_utils.surrogate_key(['event_id','device_id','client_event_time','amplitude_user_id']) }} as unique_event_id
-from final
+select *
+from surrogate
```
Comment (fivetran-joemarkiewicz) [inline models/stg_amplitude__event.sql:85]: Wanted to follow our standards and have the final statement be a `select *`. Made the relevant udpates.

File: models/stg_amplitude__event.sql [hunk 6]
```diff
@@ -22,7 +22,7 @@ final as (
         id as event_id,
         cast(event_time as {{ dbt_utils.type_timestamp() }}) as event_time,
         {{ dbt_utils.surrogate_key(['user_id','session_id']) }} as unique_session_id,
-        coalesce(user_id, (cast(amplitude_id as {{ dbt_utils.type_string() }}))) as amplitude_user_id,
+        coalesce(cast(user_id as {{ dbt_utils.type_string() }}), (cast(amplitude_id as {{ dbt_utils.type_string() }}))) as amplitude_user_id,
```
Comment (fivetran-joemarkiewicz) [inline models/stg_amplitude__event.sql:25]: My package run was failing when I didn't cast this. Updated it to address the runtime error.

File: models/stg_amplitude__event_type.sql [hunk 7]
```diff
@@ -39,13 +39,13 @@ final_pre as (
     from fields
 ),
 
-final as (
+surrogate as (
```
Comment (fivetran-joemarkiewicz) [inline models/stg_amplitude__event_type.sql:42]: Split this out into it's own cte to follow our style standard guidelines.

**Review by fivetran-joemarkiewicz — COMMENTED** [review]
@fivetran-reneeli we are getting really close!! I made a few small edits and also posed some questions to you. If my edits make sense, feel free to resolve the comment. There are just a few where a response is needed. Once those replies are addressed we can merge this PR. Additionally would you be able to:
- [x] Regenerate the docs without the `my_first` and `my_second` dbt models in the dag

**Comment by fivetran-reneeli** [comment]
Merged and regenerated docs!

**Pull request merged**